crates/rpc/rpc-engine-api/src/engine_api.rs

```diff
@@ -772,13 +751,20 @@ where
     /// Caution: This should not accept the `withdrawals` field
     async fn new_payload_v1(&self, payload: ExecutionPayloadV1) -> RpcResult<PayloadStatus> {
         trace!(target: "rpc::engine", "Serving engine_newPayloadV1");
+        let payload =
+            ExecutionData { payload: payload.into(), sidecar: ExecutionPayloadSidecar::none() };
```

Comment:
@emhane now we bubbled all of this up to the rpc server trait impl, meaning we can reuse all of the metered fns in the OP impl out of the box